,dep,version,filepath,bloated,cve,affected versions,advisory
0,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2014-3589,"<2.3.2,>=2.5.0,<2.5.2","Pillow versions 2.3.2 and 2.5.2 include a fix for CVE-2014-3589: PIL/IcnsImagePlugin.py in Python Imaging Library (PIL) and Pillow before 2.3.2 and 2.5.x before 2.5.2 allows remote attackers to cause a denial of service via a crafted block size.
https://github.com/python-pillow/Pillow/commit/205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d"
1,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2020-15999,<8.0.1,Pillow 8.0.1 updates 'FreeType' used in binary wheels to v2.10.4 to include a security fix.
2,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2020-35654,<8.1.0,Pillow 8.1.0 fixes TIFF OOB Write error. CVE-2020-35654 #5175.
3,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2020-35653,<8.1.0,"In Pillow before 8.1.0, PcxDecode has a buffer over-read when decoding a crafted PCX file because the user-supplied stride value is trusted for buffer calculations."
4,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2020-35655,<8.1.0,Pillow 8.1.0 includes a fix for SGI Decode buffer overrun. CVE-2020-35655 #5173.
5,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-25289,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25289: TiffDecode has a heap-based buffer overflow when decoding crafted YCbCr files because of certain interpretation conflicts with LibTIFF in RGBA mode. NOTE: this issue exists because of an incomplete fix for CVE-2020-35654.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
6,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-25290,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25290: In TiffDecode.c, there is a negative-offset memcpy with an invalid size.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
7,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-27922,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-27922: Pillow before 8.1.1 allows attackers to cause a denial of service (memory consumption) because the reported size of a contained image is not properly checked for an ICNS container, and thus an attempted memory allocation can be very large.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
8,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-25291,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25291: In TiffDecode.c, there is an out-of-bounds read in TiffreadRGBATile via invalid tile boundaries.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
9,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-27921,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-27921: Pillow before 8.1.1 allows attackers to cause a denial of service (memory consumption) because the reported size of a contained image is not properly checked for a BLP container, and thus an attempted memory allocation can be very large.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
10,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-25293,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25293: There is an out-of-bounds read in SGIRleDecode.c.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
11,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-25292,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25292: The PDF parser allows a regular expression DoS (ReDoS) attack via a crafted PDF file because of a catastrophic backtracking regex.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
12,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-25288,<8.2.0,"Pillow 8.2.0 includes a fix for CVE-2021-25288: There is an out-of-bounds read in J2kDecode, in j2ku_gray_i.
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-25287-cve-2021-25288-fix-oob-read-in-jpeg2kdecode"
13,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-28677,<8.2.0,"Pillow version 8.2.0 includes a fix for CVE-2021-28677: For EPS data, the readline implementation used in EPSImageFile has to deal with any combination of \r and \n as line endings. It used an accidentally quadratic method of accumulating lines while looking for a line ending. A malicious EPS file could use this to perform a DoS of Pillow in the open phase, before an image was accepted for opening.
https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MQHA5HAIBOYI3R6HDWCLAGFTIQP767FL/
https://github.com/python-pillow/Pillow/pull/5377
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-28677-fix-eps-dos-on-open"
14,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-28678,<8.2.0,"Pillow version 8.2.0 includes a fix for CVE-2021-28678: For BLP data, BlpImagePlugin did not properly check that reads (after jumping to file offsets) returned data. This could lead to a DoS where the decoder could be run a large number of times on empty data.
https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MQHA5HAIBOYI3R6HDWCLAGFTIQP767FL/
https://github.com/python-pillow/Pillow/pull/5377
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-28678-fix-blp-dos"
15,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-28676,<8.2.0,"Pillow version 8.2.0 includes a fix for CVE-2021-28676: For FLI data, FliDecode did not properly check that the block advance was non-zero, potentially leading to an infinite loop on load.
https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MQHA5HAIBOYI3R6HDWCLAGFTIQP767FL/
https://github.com/python-pillow/Pillow/pull/5377
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-28676-fix-fli-dos"
16,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-25287,<8.2.0,"Pillow 8.2.0 includes a fix for CVE-2021-25287: There is an out-of-bounds read in J2kDecode, in j2ku_graya_la.
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-25287-cve-2021-25288-fix-oob-read-in-jpeg2kdecode"
17,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-34552,<8.3.0,"Pillow 8.3.0 includes a fix for CVE-2021-34552: Pillow through 8.2.0 and PIL (also known as Python Imaging Library) through 1.1.7 allow an attacker to pass controlled parameters directly into a convert function to trigger a buffer overflow in Convert.c
https://pillow.readthedocs.io/en/stable/releasenotes/8.3.0.html#buffer-overflow
https://pillow.readthedocs.io/en/stable/releasenotes/index.html"
18,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-22816,<9.0.0,"Pillow 9.0.0 includes a fix for CVE-2022-22816: path_getbbox in path.c in Pillow before 9.0.0 has a buffer over-read during initialization of ImagePath.Path.
https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling"
19,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,PVE-2021-44525,<9.0.0,"Pillow 9.0.0 excludes carriage return in PDF regex to help prevent ReDoS.
https://github.com/python-pillow/Pillow/pull/5912
https://github.com/python-pillow/Pillow/commit/43b800d933c996226e4d7df00c33fcbe46d97363"
20,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,PVE-2022-44524,<9.0.0,"Pillow 9.0.0 ensures JpegImagePlugin stops at the end of a truncated file to avoid Denial of Service attacks.
https://github.com/python-pillow/Pillow/pull/5921
https://github.com/advisories/GHSA-4fx9-vc88-q2xc"
21,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-22815,<9.0.0,"Pillow 9.0.0 includes a fix for CVE-2022-22815: path_getbbox in path.c in Pillow before 9.0.0 improperly initializes ImagePath.Path.
https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling"
22,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-24303,<9.0.1,Pillow before 9.0.1 allows attackers to delete files because spaces in temporary pathnames are mishandled.
23,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-22817,<9.0.1,"Pillow 9.0.1 includes a fix for CVE-2022-22817: PIL.ImageMath.eval in Pillow before 9.0.0 allows evaluation of arbitrary expressions, such as ones that use the Python exec method. A first patch was issued for version 9.0.0 but it did not prevent builtins available to lambda expressions.
https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security"
24,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-45198,<9.2.0,Pillow before 9.2.0 performs Improper Handling of Highly Compressed GIF Data (Data Amplification).
25,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-28675,">=0,<8.2.0",An issue was discovered in Pillow before 8.2.0. PSDImagePlugin.PsdImageFile lacked a sanity check on the number of input layers relative to the size of the data block. This could lead to a DoS on Image.open prior to Image.load.
26,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-27923,">=4.3.0,<8.1.1","Pillow before 8.1.1 allows attackers to cause a denial of service (memory consumption) because the reported size of a contained image is not properly checked for an ICO container, and thus an attempted memory allocation can be very large."
27,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-23437,">=5.2.0,<8.3.2","Pillow from 5.2.0 and before 8.3.2 is vulnerable to Regular Expression Denial of Service (ReDoS) via the getrgb function.
https://github.com/python-pillow/Pillow/commit/9e08eb8f78fdfd2f476e1b20b7cf38683754866b
https://pillow.readthedocs.io/en/stable/releasenotes/8.3.2.html"
28,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-30595,">=9.1.0,<9.1.1",libImaging/TgaRleDecode.c in Pillow 9.1.0 has a heap buffer overflow in the processing of invalid TGA image files.
29,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,PVE-2023-55182,">=9.1.0,<9.3.0","Pillow 9.3.0 includes a security fix: Pillow will now decode the data in its natural CMYK mode, then convert it to RGB and rearrange the channels afterwards. Trying to load the data in an incorrect mode could result in a segmentation fault.
https://pillow.readthedocs.io/en/stable/releasenotes/9.3.0.html#decode-jpeg-compressed-blp1-data-in-original-mode"
30,Pillow,>=7.1.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-45199,">=9.2.0,<9.3.0","Pillow before 9.3.0 allows denial of service via SAMPLESPERPIXEL.
https://pillow.readthedocs.io/en/stable/releasenotes/9.3.0.html#limit-samplesperpixel-to-avoid-runtime-dos"
31,matplotlib,>=3.3,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
32,gitpython,>=3.1.30,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
33,tqdm,>=4.64.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
34,torchvision,>=0.8.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
35,numpy,>=1.18.5,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-33430,<1.21.0rc1,"Numpy 1.21.0rc1 includes a fix for CVE-2021-33430: A Buffer Overflow vulnerability in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. 
NOTE: The vendor does not agree this is a vulnerability. In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.
https://github.com/numpy/numpy/issues/18939"
36,numpy,>=1.18.5,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-34141,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-34141: An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. 
NOTE: the vendor states that this reported code behavior is ""completely harmless.""
https://github.com/numpy/numpy/issues/18993"
37,numpy,>=1.18.5,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-41496,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-41496: Buffer overflow in the array_from_pyobj function of fortranobject.c, which allows attackers to conduct a Denial of Service attacks by carefully constructing an array with negative values. 
NOTE: The vendor does not agree this is a vulnerability; the negative dimensions can only be created by an already privileged user (or internally).
https://github.com/numpy/numpy/issues/19000"
38,numpy,>=1.18.5,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2021-41495,<1.22.2,"Numpy 1.22.2  includes a fix for CVE-2021-41495: Null Pointer Dereference vulnerability exists in numpy.sort in NumPy in the PyArray_DescrNew function due to missing return-value validation, which allows attackers to conduct DoS attacks by repetitively creating sort arrays. 
NOTE: While correct that validation is missing, an error can only occur due to an exhaustion of memory. If the user can exhaust memory, they are already privileged. Further, it should be practically impossible to construct an attack which can target the memory exhaustion to occur at exactly this place.
https://github.com/numpy/numpy/issues/19038"
39,PyYAML,>=5.3.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2020-14343,<5.4,"Pyyaml version 5.4 includes a fix for CVE-2020-14343: A vulnerability was discovered in the PyYAML library in versions before 5.4, where it is susceptible to arbitrary code execution when it processes untrusted YAML files through the full_load method or with the FullLoader loader. Applications that use the library to process untrusted input may be vulnerable to this flaw. This flaw allows an attacker to execute arbitrary code on the system by abusing the python/object/new constructor. This flaw is due to an incomplete fix for CVE-2020-1747.
https://bugzilla.redhat.com/show_bug.cgi?id=1860466"
40,PyYAML,>=5.3.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-20477,">=5.1,<=5.1.2","PyYAML 5.1 through 5.1.2 has insufficient restrictions on the load and load_all functions because of a class deserialization issue, e.g., Popen is a class in the subprocess module. See CVE-2019-20477. NOTE: this issue exists because of an incomplete fix for CVE-2017-18342."
41,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-7317,"<3.4.18,>=4.0.0,<4.6.0",Opencv-python 3.4.18 and 4.6.0 pin its dependency 'libpng' to v1.6.37 to include a security fix.
42,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-0778,"<3.4.18,>=4.0.0,<4.6.0",Opencv-python 3.4.18 and 4.6.0 update its dependency 'openssl' to v1_1_1o to include security fixes.
43,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-1292,"<3.4.18,>=4.0.0,<4.6.0",Opencv-python 3.4.18 and 4.6.0 update its dependency 'openssl' to v1_1_1o to include security fixes.
44,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-27406,"<3.4.18,>=4.0.0,<4.6.0",Opencv-python 3.4.18 and 4.6.0 pin its dependency 'freetype' to v2.12.1 to include a security fix.
45,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-9423,<4.1.1.26,"Opencv-python 4.1.1.26 includes a fix for CVE-2019-9423: In opencv calls that use libpng, there is a possible out of bounds write due to a missing bounds check. This could lead to local escalation of privilege with no additional execution privileges required. User interaction is not required for exploitation."
46,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-15939,<4.1.1.26,Opencv-python 4.1.1.26 includes a fix for CVE-2019-15939: An issue was discovered in OpenCV 4.1.0. There is a divide-by-zero error in cv::HOGDescriptor::getDescriptorSize in modules/objdetect/src/hog.cpp.
47,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-16249,<4.1.2.30,Opencv-python 4.1.2.30 includes a fix for  CVE-2019-16249: OpenCV 4.1.1 has an out-of-bounds read in hal_baseline::v_load in core/hal/intrin_sse.hpp when called from computeSSDMeanNorm in modules/video/src/dis_flow.cpp.
48,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-37434,<4.7.0,Opencv-python 4.7.0 updates its C dependency 'zlib' to v1.2.13 to include a security fix.
49,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-2097,<4.7.0,Opencv-python 4.7.0 updates its dependency 'OpenSSL' to v1.1.1s to include security fixes.
50,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-1475,<4.7.0,Opencv-python 4.7.0 updates its C dependency 'FFmpeg' to v5.1.2 to include a security fix.
51,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-2068,<4.7.0,Opencv-python 4.7.0 updates its dependency 'OpenSSL' to v1.1.1s to include security fixes.
52,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-12864,">=0,<3.3.1.11","In opencv/modules/imgcodecs/src/grfmt_pxm.cpp, function ReadNumber did not checkout the input length, which lead to integer overflow. If the image is from remote, may lead to remote code execution or denial of service. This affects OpenCV 3.3 (corresponding with OpenCV-Python version 3.3.0.9) and earlier."
53,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-12605,">=0,<3.3.1.11",OpenCV (Open Source Computer Vision Library) through 3.3 (corresponding to OpenCV-Python and OpenCV-Contrib-Python 3.3.0.9) has an out-of-bounds write error in the FillColorRow8 function in utils.cpp when reading an image file by using cv::imread.
54,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-1000450,">=0,<3.3.1.11","In opencv/modules/imgcodecs/src/utils.cpp, functions FillUniColor and FillUniGray do not check the input length, which can lead to integer overflow. If the image is from remote, may lead to remote code execution or denial of service. This affects Opencv 3.3 (corresponding with OpenCV-Python 3.3.0.9) and earlier."
55,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-12863,">=0,<3.3.1.11","In opencv/modules/imgcodecs/src/grfmt_pxm.cpp, function PxMDecoder::readData has an integer overflow when calculate src_pitch. If the image is from remote, may lead to remote code execution or denial of service. This affects OpenCV 3.3 (corresponding to OpenCV-Python 3.3.0.9) and earlier."
56,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2016-1516,">=0,<3.3.1.11",OpenCV 3.0.0 has a double free issue that allows attackers to execute arbitrary code. This issue was fixed in OpenCV version 3.3.1 (corresponding to OpenCV-Python and and OpenCV-Contrib-Python 3.3.1.11).
57,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-14136,">=0,<3.3.1.11",OpenCV (Open Source Computer Vision Library) 3.3 (corresponding to OpenCV-Python 3.3.0.9) has an out-of-bounds write error in the function FillColorRow1 in utils.cpp when reading an image file by using cv::imread. NOTE: this vulnerability exists because of an incomplete fix for CVE-2017-12597.
58,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-12862,">=0,<3.3.1.11","In modules/imgcodecs/src/grfmt_pxm.cpp, the length of buffer AutoBuffer _src is small than expected, which will cause copy buffer overflow later. If the image is from remote, may lead to remote code execution or denial of service. This affects OpenCV 3.3 (corresponding to OpenCV-Python 3.3.0.9) and earlier."
59,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-17760,">=0,<3.4.0.12","OpenCV 3.3.1 (corresponding with opencv-python and opencv-contrib-python 3.3.1.11) has a Buffer Overflow in the cv::PxMDecoder::readData function in grfmt_pxm.cpp, because an incorrect size value is used."
60,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2017-18009,">=0,<3.4.1.15","In OpenCV 3.3.1 (corresponding with OpenCV-Python 3.3.1.11), a heap-based buffer over-read exists in the function cv::HdrDecoder::checkSignature in modules/imgcodecs/src/grfmt_hdr.cpp."
61,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2018-5268,">=0,<3.4.1.15","In OpenCV 3.3.1 (corresponding with OpenCV-Python 3.3.1.11), a heap-based buffer overflow happens in cv::Jpeg2KDecoder::readComponent8u in modules/imgcodecs/src/grfmt_jpeg2000.cpp when parsing a crafted image file."
62,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2018-5269,">=0,<3.4.1.15","In OpenCV 3.3.1 (corresponds with OpenCV-Python 3.3.1.11), an assertion failure happens in cv::RBaseStream::setPos in modules/imgcodecs/src/bitstrm.cpp because of an incorrect integer cast."
63,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-14493,">=0,<4.1.1.26",An issue was discovered in OpenCV before 4.1.1 (OpenCV-Python before 4.1.1.26). There is a NULL pointer dereference in the function cv::XMLParser::parse at modules/core/src/persistence.cpp.
64,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-14492,">=4.0.0.21,<4.1.1.26,<3.4.7.28","Opencv-python 4.1.1.26 and 3.4.7.28 include a fix for CVE-2019-14492: An issue was discovered in OpenCV before 3.4.7 and 4.x before 4.1.1. There is an out of bounds read/write in the function HaarEvaluator::OptFeature::calc in modules/objdetect/src/cascadedetect.hpp, which leads to denial of service."
65,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-14491,">=4.0.0.21,<4.1.1.26,<3.4.7.28","Opencv-python 4.1.1.26 and 3.4.7.28 include a fix for CVE-2019-14491: An issue was discovered in OpenCV before 3.4.7 and 4.x before 4.1.1. There is an out of bounds read in the function cv::predictOrdered<cv::HaarEvaluator> in modules/objdetect/src/cascadedetect.hpp, which leads to denial of service."
66,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-5063,">=4.0.0.21,<4.2.0.32","Opencv-python 4.2.0.32 includes a fix for CVE-2019-5063: An exploitable heap buffer overflow vulnerability exists in the data structure persistence functionality of OpenCV 4.1.0. A specially crafted XML file can cause a buffer overflow, resulting in multiple heap corruptions and potential code execution. An attacker can provide a specially crafted file to trigger this vulnerability."
67,opencv-python,>=4.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-5064,">=4.0.0.21,<4.2.0.32","Opencv-python 4.2.0.32 includes a fix for  CVE-2019-5064: An exploitable heap buffer overflow vulnerability exists in the data structure persistence functionality of OpenCV, before version 4.2.0. A specially crafted JSON file can cause a buffer overflow, resulting in multiple heap corruptions and potentially code execution. An attacker can provide a specially crafted file to trigger this vulnerability."
68,requests,>=2.23.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2015-2296,">=2.1,<=2.5.3",The resolve_redirects function in sessions.py in requests 2.1.0 through 2.5.3 allows remote attackers to conduct session fixation attacks via a cookie without a host value in a redirect.
69,requests,>=2.23.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2023-32681,">=2.3.0,<2.31.0","Requests is a HTTP library. Since Requests 2.3.0, Requests has been leaking Proxy-Authorization headers to destination servers when redirected to an HTTPS endpoint. This is a product of how we use 'rebuild_proxies' to reattach the 'Proxy-Authorization' header to requests. For HTTP connections sent through the tunnel, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the 'Proxy-Authorization' header must be sent in the CONNECT request as the proxy has no visibility into the tunneled request. This results in Requests forwarding proxy credentials to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate sensitive information. This issue has been patched in version 2.31.0.

"
70,ultralytics,>=8.0.145,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
71,seaborn,>=0.11.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
72,psutil,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2019-18874,<=5.6.5,psutil (aka python-psutil) through 5.6.5 can have a double free. This occurs because of refcount mishandling within a while or for loop that converts system data into a Python object. See CVE-2019-18874.
73,thop,>=0.1.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
74,pandas,>=1.1.4,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2020-13091,">=0,<1.0.4","** DISPUTED ** pandas through 1.0.3 can unserialize and execute commands from an untrusted file that is passed to the read_pickle() function, if __reduce__ makes an os.system call. NOTE: third parties dispute this issue because the read_pickle() function is documented as unsafe and it is the user's responsibility to use the function in a secure manner."
75,setuptools,>=65.5.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,no one,no one,no one
76,scipy,>=1.4.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2013-4251,">=0,<0.12.1",The scipy.weave component in SciPy before 0.12.1 creates insecure temporary directories.
77,torch,>=1.7.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/requirements.txt,False,CVE-2022-45907,">=0,<1.13.1","Torch 1.13.1 includes a fix for CVE-2022-45907: In PyTorch before trunk/89695, torch.jit.annotations.parse_type_line can cause arbitrary code execution because eval is used unsafely.
https://github.com/pytorch/pytorch/issues/88868"
78,werkzeug,>=2.2.3,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/utils/google_app_engine/additional_requirements.txt,False,CVE-2019-14806,">=0,<0.15.3","Pallets Werkzeug before 0.15.3, when used with Docker, has insufficient debugger PIN randomness because Docker containers share the same machine id."
79,werkzeug,>=2.2.3,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/utils/google_app_engine/additional_requirements.txt,False,CVE-2019-14322,">=0,<0.15.5","In Pallets Werkzeug before 0.15.5, SharedDataMiddleware mishandles drive names (such as C:) in Windows pathnames."
80,werkzeug,>=2.2.3,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/utils/google_app_engine/additional_requirements.txt,False,CVE-2022-29361,">=0,<2.1.1",** DISPUTED ** Improper parsing of HTTP requests in Pallets Werkzeug v2.1.0 and below allows attackers to perform HTTP Request Smuggling using a crafted HTTP request with multiple requests included inside the body. NOTE: the vendor's position is that this behavior can only occur in unsupported configurations involving development mode and an HTTP server from outside the Werkzeug project.
81,werkzeug,>=2.2.3,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/utils/google_app_engine/additional_requirements.txt,False,PVE-2021-37276,">=0.15.0,<0.15.5",:class:`~exceptions.BadRequestKeyError` in werkzeug 0.15.5 adds the ``KeyError`` message to the description if ``e.show_exception`` is set to ``True``. This is a more secure default than the original 0.15.0 behavior and makes it easier to control without losing information.
82,pip,==21.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/utils/google_app_engine/additional_requirements.txt,False,no one,no one,no one
83,Flask,==2.3.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/utils/google_app_engine/additional_requirements.txt,False,CVE-2023-30861,"<2.2.5,>=2.3.0,<2.3.2","Flask is a lightweight WSGI web application framework. When all of the following conditions are met, a response containing data intended for one client may be cached and subsequently sent by the proxy to other clients. If the proxy also caches 'Set-Cookie' headers, it may send one client's 'session' cookie to other clients. The severity depends on the application's use of the session and the proxy's behavior regarding cookies. The risk depends on all these conditions being met.

1. The application must be hosted behind a caching proxy that does not strip cookies or ignore responses with cookies.
2. The application sets 'session.permanent = True'
3. The application does not access or modify the session at any point during a request.
4. 'SESSION_REFRESH_EACH_REQUEST' enabled (the default).
5. The application does not set a 'Cache-Control' header to indicate that a page is private or should not be cached.

This happens because vulnerable versions of Flask only set the 'Vary: Cookie' header when the session is accessed or modified, not when it is refreshed (re-sent to update the expiration) without being accessed or modified. This issue has been fixed in versions 2.3.2 and 2.2.5."
84,gunicorn,==19.10.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/yolov5/utils/google_app_engine/additional_requirements.txt,False,PVE-2021-40104,"<19.10.0,>=20.0.0,<20.0.1","Gunicorn 20.0.1 fixes chunked encoding support to prevent http request smuggling attacks.
https://github.com/benoitc/gunicorn/issues/2176
https://github.com/p4k03n4t0r/http-request-smuggling#request-smuggling-using-mitmproxy-and-gunicorn"
